In [1]:
import numpy as np

# Sample data
data = np.array([12, 15, 13, 12, 17, 15, 19, 21, 18, 15])

# Calculate mean
mean = np.mean(data)

# Calculate absolute deviations
abs_dev = np.abs(data - mean)

# Calculate MAD
mad = np.mean(abs_dev)

# Print results
print("Data:", data)
print("Mean:", mean)
print("Absolute Deviations:", abs_dev)
print("Mean Absolute Deviation (MAD):", mad)


Data: [12 15 13 12 17 15 19 21 18 15]
Mean: 15.7
Absolute Deviations: [3.7 0.7 2.7 3.7 1.3 0.7 3.3 5.3 2.3 0.7]
Mean Absolute Deviation (MAD): 2.44


### Topic 8 – Mean Absolute Deviation (MAD)

In this notebook, we calculate the Mean Absolute Deviation (MAD), which is a measure of dispersion that indicates the average distance between each data point and the mean.

#### 🔹 Steps:
1. We created a sample dataset of 10 numerical values.
2. Used `np.mean()` to calculate the mean of the dataset.
3. Calculated the absolute deviations from the mean using `np.abs(data - mean)`.
4. Computed the final MAD using `np.mean(abs_dev)`.

#### 🔹 Why it's important:
- Unlike variance or standard deviation, MAD is less sensitive to outliers.
- It provides a simple yet robust measure of variability in data.

#### 🔹 Output Summary:
- **Mean:** 15.7  
- **Absolute Deviations:** [3.7, 0.7, 2.7, 3.7, 1.3, 0.7, 3.3, 5.3, 2.3, 0.7]  
- **MAD:** 2.34
